### ISSUE TITLE
bug(cdkTrapFocus): breaks in safari if buttons are contenteditable and user-select:auto

### ISSUE THREAD

**Issue by Frazer** [angular/components#23846]
#### Reproduction

Use StackBlitz to reproduce your issue: 
https://stackblitz.com/edit/components-issue-gnh425?file=src%2Fapp%2Fexample-component.html
(sorry I don't know how to fix the 'ngcc failed to run on @angular/cdk' error)

Steps to reproduce:
1.  In Safari, click the button to open the dialog
2. Push tab to move between buttons on the dialog
3. Attempt to paste to the dialog

Remove user-select: auto; from the dialog
 

1.  In Safari, click the button to open the dialog
2. Push tab to move between buttons on the dialog
3. Attempt to paste to the dialog

 
#### Expected Behavior

The dialog opens. The focus is captured in the dialog. Pasting triggers the onPaste function.

#### Actual Behavior

With user-select: auto; the focus is not captured in the dialog.
Without user-select: auto; the ability to paste to the dialog does not work while the buttons have focus.

Works as expected with or without user-select on Chrome, Firefox, Edge

#### Environment

- Angular:  8.0.0
- CDK/Material: 8.0.0
- Browser(s):  Safari
- Operating System (e.g. Windows, macOS, Ubuntu):  Macs

**Comment by crisbeto** [comment]
We're in the process of cleaning up our backlog. I'm closing this issue, because we can't do much about it, as far as I can tell. This looks like something that would be fixed in the browser.
